src/Domain/class-wc-email-customer-new-account-blocks.php

```diff
@@ -0,0 +1,190 @@
+<?php
+/**
+ * Class WC_Email_Customer_New_Account_Blocks file.
+ *
+ * @package  WooCommerce/Blocks
+ */
+
+use WC_Email;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+if ( ! class_exists( 'WC_Email_Customer_New_Account_Blocks', false ) ) :
+
+	/**
+	 * Customer New Account.
+	 *
+	 * An email sent to the customer when they create an account.
+	 * This is intended as a replacement to WC_Email_Customer_New_Account(),
+	 * with a set password link instead of emailing the new password in email
+	 * content.
+	 *
+	 * @extends     WC_Email
+	 */
+	class WC_Email_Customer_New_Account_Blocks extends \WC_Email {
+
+		/**
+		 * User login name.
+		 *
+		 * @var string
+		 */
+		public $user_login;
+
+		/**
+		 * User email.
+		 *
+		 * @var string
+		 */
+		public $user_email;
+
+		/**
+		 * Magic link to set initial password.
+		 *
+		 * @var string
+		 */
+		public $set_password_url;
+
+		/**
+		 * Override (force) default template path
+		 *
+		 * @var string
+		 */
+		public $default_template_path;
+
+		/**
+		 * Constructor.
+		 */
+		public function __construct() {
+			// Note - we're using the same ID as the real email.
+			// This ensures that any merchant tweaks (Settings > Emails)
+			// apply to this email (consistent with the core email).
+			$this->id                    = 'customer_new_account';
+			$this->customer_email        = true;
+			$this->title                 = __( 'New account', 'woo-gutenberg-products-block' );
+			$this->description           = __( 'Customer "new account" emails are sent to the customer when a customer signs up via checkout or account blocks.', 'woo-gutenberg-products-block' );
+			$this->template_html         = 'emails/customer-new-account-blocks.php';
+			$this->template_plain        = 'emails/plain/customer-new-account-blocks.php';
+			$this->default_template_path = plugin_dir_path( __FILE__ ) . 'templates/';
```

Comment:
Let's chat about this, I think we'll need to use some dependency injection here but there's bunch of ways of doing it so we should hash it out a bit before committing. Ideally ALL paths will come from `Automattic\WooCommerce\Blocks\Domain\Package`.  The `templates` folder should also be at `src/templates` I think (or even `/templates`).

Reply:
Took a stab at this - 648a572a236ad8801b7e9088e9f07f4031757fe0 

Hopefully a good start, there's a tricky dependency on `WC_Email` from woo core. `Package::get_path` was definitely what I was looking for 🏞️